### ISSUE TITLE
[Panel] Forbidden user roles still selectable in user create dialog

### ISSUE THREAD

**Issue by texnixe** [getkirby/kirby#1917]
**Describe the bug**
If I limit what roles a user can create via the options setting:

```
title: Editor

options:
  create:
    editor: false
    manager: true
```

The dialog still shows all roles as selectable:

<img width="482" alt="image" src="https://user-images.githubusercontent.com/5312492/61183827-36462900-a646-11e9-9d4a-6466ba312015.png">

**To Reproduce**
1. Create an Editor and a Manager role blueprint, in the Editor blueprint add the above options
2. Create an Editor user and log in with that user
3. Click on "Add new user"
4. The above dialog is opened.
5. Select Editor as role: you get an error message; if you change role to manager, the user is created.

**Expected behavior**
The user roles(s) that cannot be created shouldn't be selectable in the first place.

**Kirby Version**
3.2.2

**Additional context**

**Comment by texnixe** [comment]
On a side note, I find it confusing to have permissions and  options in user blueprints with the same sort of properties. Why this additional options setting instead of putting that into the permissions sections?

```
title: Editor

permissions:
  users:
    create:
      editor: false
      manager: true
```

(This doesn't work though)

**Comment by texnixe** [comment]
See also: https://github.com/getkirby/kirby/issues/2038

**Comment by distantnative** [comment]
I am majorly confused after running in circles:
I don't think we support an `option` setting in user blueprints. Only `permissions`. Or do we?

**Comment by texnixe** [comment]
We do, but I didn't really get it when I posted this issue. Here is what Bastian wrote on Slack:

> in users/editor.yml sind die Permissions dafür da zu Reglen, was Editoren allgemein im System dürfen.
Options regeln hingegen, was passiert wenn ein User mit einer anderen Rolle einen Editor bearbeitet.

**Comment by distantnative** [comment]
**Developer's note:**
What I am currently struggling with: the options should be tied to the role object to filter them in the Panel accordingly. However, in our setup the options in a role user blueprint end up with the user... which does not make that much sense, does it - @bastianallgeier?

**Comment by bastianallgeier** [comment]
I think I solved this in #2108. Please have a look at it and let me know if this makes any sense to you :)

**Comment by bastianallgeier** [comment]
✅
